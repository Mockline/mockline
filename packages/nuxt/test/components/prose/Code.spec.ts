import { describe, it, expect } from 'vitest'
// @ts-ignore
import { MCode } from '#components'
import type { TypeOf } from "zod";
import ComponentRender from "../../component-render";

describe('Code', () => {
  it.each([
    ['basic case', {}],
    ['with lang', { props: { lang: 'typescript' }}],
  ])('renders %s correctly', async (nameOrHtml, options: TypeOf<typeof MCode.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MCode)
    expect(html).toMatchSnapshot()
  })
})
