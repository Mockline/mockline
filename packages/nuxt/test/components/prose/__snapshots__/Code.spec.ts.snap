// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Code > renders basic case correctly 1`] = `"<code class="px-1.5 py-0.5 text-sm font-mono font-medium rounded-base inline-block border border-[var(--color-border-primary)] text-[var(--color-text-primary)] bg-[var(--color-bg-level-2)]"></code>"`;

exports[`Code > renders with lang correctly 1`] = `"<code class="px-1.5 py-0.5 text-sm font-mono font-medium rounded-base inline-block border border-[var(--color-border-primary)] text-[var(--color-text-primary)] bg-[var(--color-bg-level-2)]"></code>"`;
